src/Compilers/CSharp/Portable/Symbols/Synthesized/Extensions/SynthesizedExtensionMarker.cs

```diff
@@ -0,0 +1,137 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    /// <summary>
+    /// This method encodes the information needed to round-trip extension types
+    /// through metadata.
+    ///
+    /// It encodes:
+    /// - whether the extension type is implicit or explicit (PROTOTYPE)
+    /// - the underlying type (first parameter type)
+    /// - the base extensions (subsequent parameter types)
+    /// </summary>
+    internal sealed class SynthesizedExtensionMarker : SourceOrdinaryMethodSymbolBase
```

Comment:
>SourceOrdinaryMethodSymbolBase [](http://example.com/codeflow?start=55&length=30)

I do not think this should be a source symbol

Reply:
If we are taking advantage of this base class in a particular way, it would be good to clearly document the fact.